x-pack/plugins/monitoring/public/application/hooks/use_clusters.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useState, useEffect } from 'react';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { STANDALONE_CLUSTER_CLUSTER_UUID } from '../../../common/constants';
+
+export function useClusters(codePaths?: string[], fetchAllClusters?: boolean, ccs?: any) {
```

Comment:
This is the initial migration for the [cluster.js service ](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/public/services/clusters.js) and the [route_init.js](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/public/lib/route_init.js) file.
It is not completed so is going to evolve as we migrate more routes and use cases. I thought it didn't make sense to migrate it completely as we would need more views to migrate to test it works as before.